File: app/src/main/java/com/parassidhu/pdfpin/DataAdapter.java
package com.parassidhu.pdfpin;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import com.yarolegovich.lovelydialog.LovelyTextInputDialog;

import java.util.ArrayList;

public class DataAdapter extends RecyclerView.Adapter<DataAdapter.ViewHolder> {
    private ArrayList<ListItem> listItems;
    private Context context;
    int i =0;

    public DataAdapter(Context context,ArrayList<ListItem> listItems) {
        this.context = context;
        this.listItems = listItems;
    }

    @Override
    public DataAdapter.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.row_layout, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final ViewHolder viewHolder, int i) {
        viewHolder.filename.setText(listItems.get(i).getName());
        viewHolder.edit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new LovelyTextInputDialog(context)
                        .setTopColorRes(R.color.blue)
                        .setTitle("Rename")
                        .setInitialInput(viewHolder.filename.getText().toString())
                        .setConfirmButton(android.R.string.ok, new LovelyTextInputDialog.OnTextInputConfirmListener() {
                            @Override
                            public void onTextInputConfirmed(String text) {
                                if(!text.isEmpty()) {
                                    listItems.get(viewHolder.getAdapterPosition()).setName(text);
                                    viewHolder.filename.setText(listItems.get(viewHolder.getAdapterPosition()).getName());
                                }
                            }
                        })
                        .show();
            }
        });
    }

    @Override
    public int getItemCount() {
        return listItems.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView filename;
        ImageButton edit;
        public ViewHolder(View view) {
            super(view);
            filename = (TextView)view.findViewById(R.id.filename);
            edit = (ImageButton) view.findViewById(R.id.edit);
        }
    }
}

File: app/src/main/java/com/parassidhu/pdfpin/DataAdapter.kt
package com.parassidhu.pdfpin

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.yarolegovich.lovelydialog.LovelyTextInputDialog
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.row_layout.*
import java.util.*

class DataAdapter(private val context: Context, private val listItems: ArrayList<ListItem>)
    : RecyclerView.Adapter<DataAdapter.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): DataAdapter.ViewHolder {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.row_layout, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        viewHolder.bind(i)
    }

    override fun getItemCount(): Int {
        return listItems.size
    }

    inner class ViewHolder(override val containerView: View?)
        : RecyclerView.ViewHolder(containerView!!), LayoutContainer {

        fun bind(i: Int) {
            filename.text = listItems[i].name
            edit.setOnClickListener {
                LovelyTextInputDialog(context)
                        .setTopColorRes(R.color.blue)
                        .setTitle("Rename")
                        .setInitialInput(filename.text.toString())
                        .setConfirmButton(android.R.string.ok) { text ->
                            if (!text.isEmpty()) {
                                listItems[adapterPosition].name = text
                                filename.text = listItems[adapterPosition].name
                            }
                        }
                        .show()
            }
        }
    }
}